### PR TITLE
Add cache steps for Cargo and pnpm to CI test workflow

### PR DESCRIPTION
**Pull request by CDThomas**
This PR adds cache steps for Cargo and pnpm to the CI test workflow.

This seems to shave a few min off for builds with cache hits because of fewer deps to download + only compiling stash-rs twice rather than three times.

Example of a build without the cache: https://github.com/cipherstash/cipherstash.js/runs/7800462761?check_suite_focus=true

Example of a build with the cache: https://github.com/cipherstash/cipherstash.js/runs/7800462761?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,12 +19,38 @@ jobs:
...
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+
+    - name: Get pnpm store directory
```
Comment (CDThomas) [inline .github/workflows/test.yml:33]: Copied from https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -19,12 +19,38 @@ jobs:
 
     - uses: ./.github/actions/cache-asdf
 
+    - name: "Set up Cargo cache"
```
Comment (CDThomas) [inline .github/workflows/test.yml:22]: Copied from StashRS (https://github.com/cipherstash/cipherstash-rs/blob/f9507360747368a8b7b313b1cd922eb5ebe596db/.github/workflows/test.yml#L27-L35)

**Review by bennetthardwick — APPROVED** [review]
Nice - might be worth using something like this since cargo caches grow indefinitely: https://crates.io/crates/cargo-cache

Probably won't be an issue for a while though.

**Comment by bennetthardwick** [comment]
> Nice - might be worth using something like this since cargo caches grow indefinitely: https://crates.io/crates/cargo-cache
> 
> Probably won't be an issue for a while though.

Actually don't listen to me - the whole cache gets evicted every time the cargo toml changes anyway.

**Pull request merged**